src/commands/importDocuments.ts

```diff
@@ -0,0 +1,127 @@
+    }
+    let result: string;
+    if (collectionNode.treeItem instanceof MongoCollectionTreeItem) {
+        const collectionTreeItem = <MongoCollectionTreeItem>collectionNode.treeItem;
+        result = await collectionTreeItem.executeCommand('insertMany', [JSON.stringify(documents)]);
+    } else {
+        result = await insertDocumentsIntoDocdb(<IAzureParentNode<DocDBCollectionTreeItem>>collectionNode, documents, uris);
+    }
+    await collectionNode.refresh();
+    await vscode.window.showInformationMessage(result);
+}
+
+async function askForDocuments(): Promise<vscode.Uri[]> {
+    let files: vscode.Uri[] = await vscode.workspace.findFiles("*.json");
+    let jsonDocuments: (vscode.QuickPickItem & { uri: vscode.Uri })[] = [];
+    let items: (vscode.QuickPickItem & { uri: vscode.Uri })[] = files.map(file => {
+        return { uri: file, label: vscode.workspace.asRelativePath(file) };
+    });
+    let pickAgain: string = "Pick again";
+    let discontinue = "Discontiue import";
+    while (!jsonDocuments.length) {
+        jsonDocuments = await ext.ui.showQuickPick(items, { canPickMany: true, placeHolder: "Choose a document to upload. Hit Escape to Cancel" });
+        if (!jsonDocuments.length) {
+            let action: string = await vscode.window.showWarningMessage("No document picked. Want to pick again?", pickAgain, discontinue);
+            if (action === discontinue) {
+                throw new UserCancelledError();
+            }
+        }
+    }
+    return jsonDocuments.map(choice => choice.uri);
+}
+
+// tslint:disable-next-line:no-any
+async function parseDocumentsForErrors(nodes: vscode.Uri[]): Promise<any[]> {
+    const parseResult = await parseDocuments(nodes);
+    const documents = parseResult[0];
+    const errors: string[] = parseResult[1];
+    if (errors.length > 0) {
+        throw new Error(`Errors found in the following documents: ${errors.join(',')}.\nPlease fix these and try again.`);
+    }
+    return documents;
+}
+
+// tslint:disable-next-line:no-any
+async function parseDocuments(uris: vscode.Uri[]): Promise<[any[], string[]]> {
+    let documents = [];
+    let errors = {};
+    let errorFoundFlag: boolean = false;
+    for (let uri of uris) {
+        const text: string = await fse.readFile(uri.fsPath, 'utf-8');
+        let parsed;
+        try {
+            parsed = JSON.parse(text);
+        } catch (e) {
+            if (!errorFoundFlag) {
+                ext.outputChannel.appendLine("Errors found in documents listed below. Please fix these.");
+            }
+            const err = parseError(e);
+            const fileName = uri.path.split('/').pop();
+            errors[fileName] = err;
+            ext.outputChannel.appendLine(`${fileName}:\n${err.message}`);
+            ext.outputChannel.show();
+        }
+        if (parsed) {
+            if (Array.isArray(parsed)) {
+                documents = documents.concat(parsed);
```

Comment:
This is the scenario where they've pointed you to a set of documents in JSON?  Seems like importing everything would be the right thing, no?  We can ask Matt.

It might make sense to confirm the import with a number of document that will be imported so they're not susprised…?

Reply:
Let's review the UI/experience first before making changes like this, though.

Reply:
Talked to Matt about this - we don't intend to throw errors. It is perfectly valid JSON, and if it has an array of objects, we just add them.